rs/sns/swap/src/swap.rs

```diff
@@ -3374,128 +3376,182 @@ impl SnsNeuronRecipe {
         nns_governance: CanisterId,
         sns_transaction_fee_e8s: u64,
     ) -> Result<NeuronRecipe, (ConversionError, String)> {
-        let (participant, controller) = match self.investor.as_ref() {
-            Some(Investor::Direct(DirectInvestment { buyer_principal })) => {
-                let parsed_buyer_principal = match string_to_principal(buyer_principal) {
+        let SnsNeuronRecipe {
+            sns: transferable_amount, // Mitigating a historical misnomer.
+            neuron_attributes,
+            claimed_status,
+            investor,
+        } = self;
+
+        // SnsNeuronRecipe.investor should always be present as it is set in `commit`.
+        // In the case of a bug due to programmer error, increment the invalid field.
+        // This will require a manual intervention via an upgrade to correct.
+        let investor = investor.as_ref().ok_or_else(|| {
+            (
+                ConversionError::Invalid,
+                format!("Missing investor information for neuron recipe {:?}", self),
+            )
+        })?;
+
+        // SnsNeuronRecipe.neuron_attributes should always be present as it is set in `commit`.
+        // This will require a manual intervention via an upgrade to correct
+        let neuron_attributes = neuron_attributes.as_ref().ok_or_else(|| {
+            (
+                ConversionError::Invalid,
+                format!(
+                    "Missing neuron_attributes information for neuron recipe {:?}",
+                    self
+                ),
+            )
+        })?;
+        // SnsNeuronRecipe.sns should always be present as it is set in `commit`.
+        // This will require a manual intervention via an upgrade to correct
+        let transferable_amount = transferable_amount.as_ref().ok_or_else(|| {
+            (
+                ConversionError::Invalid,
+                format!(
+                    "Missing transferable_amount (field `sns`) for neuron recipe {:?}",
+                    self
+                ),
+            )
+        })?;
+
+        // Claimed status is used for sanitization only, it does not affect the Ok result.
```

Comment:
At first it was unclear to me what this comment meant, what do yuo think of this?
```suggestion
- // Claimed status is used for sanitization only, it does not affect the Ok result.
+ // Claimed status does not influence the result of this function, other than causing it to
+ // return an Err(...) in some cases.
```